v-next/hardhat/test/internal/builtin-plugins/network-manager/json-rpc-request-modifiers/ethereum-mocked-provider.ts

```diff
@@ -0,0 +1,61 @@
+import type {
+  EIP1193Provider,
+  JsonRpcRequest,
+  JsonRpcResponse,
+  RequestArguments,
+} from "../../../../../src/types/providers.js";
+
+import EventEmitter from "node:events";
+
+// This mock is used in the unit tests to simulate the return value of the "request" method
+export class EthereumMockedProvider
```

Comment:
If we set up a way to inject a test dispatcher like I suggested in my other comment, using this mocked provider might no longer be necessary.

Reply:
See comment above